### PR TITLE
feat(docs): implement docs map generator

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Creates `generateDocsMap()` in `packages/docs/src/core/docs-map-generator.ts`
- Leverages existing `buildExpectedOutput()` to collect all doc sources, then maps them to `DocsMapEntry` records
- Infers `DocKind` from filename/path heuristics (e.g., `README.md` → `readme`, `docs/` subdir → `deep`)
- Implements read/write functions for `.outfitter/docs-map.json`
- Returns `Result<DocsMap, DocsCoreError>` following the handler contract
- Comprehensive test coverage including workspace fixture tests

## Test plan

- [ ] `generateDocsMap()` produces a valid `DocsMap` for the workspace fixture
- [ ] Kind inference correctly categorizes READMEs, deep docs, and root docs
- [ ] Output is deterministic (same input → same output)
- [ ] Read/write round-trips preserve the docs map

Closes OS-278

Closes: OS-278

<h3>Greptile Summary</h3>

Implements `generateDocsMap()` to discover and catalog markdown documentation across a monorepo workspace, producing a `.outfitter/docs-map.json` manifest with inferred document kinds and extracted titles.

- Extracts shared package discovery logic into `collectPackageDocs()` to eliminate duplication between docs-map generation and `buildExpectedOutput()`
- `inferDocKind()` classifies documents using filename/path heuristics (readme, guide, reference, architecture, convention, deep, generated, release)
- Uses token-boundary matching for API detection to avoid false positives like `rapid-deployment.md`
- Validates paths stay within workspace boundaries and detects `.md`/`.mdx` output collisions
- Comprehensive test coverage with workspace fixtures and regression tests for all previously identified issues

<h3>Confidence Score: 5/5</h3>

- Safe to merge - well-tested implementation with proper validation and no critical issues
- All previously identified issues have been addressed with fixes and regression tests. The refactoring properly shares logic, the kind inference handles edge cases, path validation is comprehensive, and test coverage is thorough including workspace fixtures and collision detection.
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/docs/src/core/docs-map-generator.ts | New docs map generator with kind inference, title extraction, and read/write functions - comprehensive validation and error handling |
| packages/docs/src/core/package-doc-collection.ts | Extracted shared package discovery logic - removes duplication between `buildExpectedOutput` and docs-map generation |
| packages/docs/src/__tests__/docs-map-generator.test.ts | Comprehensive test coverage including workspace fixtures, kind inference, mdx normalization, path validation, and collision detection |
| packages/docs/src/core/index.ts | Refactored to use shared `collectPackageDocs` helper - removed ~200 lines of duplicated logic |

</details>

</details>

<sub>Last reviewed commit: 94d22b3</sub>

### CODE REVIEW

File: packages/docs/src/core/docs-map-generator.ts [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+          id,
+          kind,
+          title,
+          sourcePath,
+          outputPath,
+          package: pkg.dirName,
+          tags: [],
+        });
+      }
+    }
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Applied via refactor. I extracted shared package-doc discovery into `collectPackageDocs` (`packages/docs/src/core/package-doc-collection.ts`), and both `buildExpectedOutput` (`packages/docs/src/core/index.ts`) and docs-map generation (`packages/docs/src/core/docs-map-generator.ts`) now call the same helper to avoid drift.

File: packages/docs/src/core/docs-map-generator.ts [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+      const sourceFiles = await collectPackageSourceFiles(
+        pkg.root,
+        excludedNames
+      );
+
+      for (const sourceAbsPath of sourceFiles) {
+        const relFromPkgRoot = relative(pkg.root, sourceAbsPath);
+        const sourcePath = relativeToWorkspace(workspaceRoot, sourceAbsPath);
+        const outputPath = toPosixPath(
+          join(outputDir, pkg.dirName, relFromPkgRoot)
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Addressed in subsequent commits on this stack; this thread is now outdated relative to the current diff. Resolving.

File: packages/docs/src/core/docs-map-generator.ts [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+ */
+export async function generateDocsMap(
+  options?: GenerateDocsMapOptions
+): Promise<Result<DocsMap, DocsCoreError>> {
+  const workspaceRoot = resolve(options?.workspaceRoot ?? process.cwd());
+  const packagesRoot = resolve(
+    workspaceRoot,
+    options?.packagesDir ?? DEFAULT_PACKAGES_DIR
+  );
+  const outputDir = options?.outputDir ?? DEFAULT_OUTPUT_DIR;
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Addressed in subsequent commits on this stack; this thread is now outdated relative to the current diff. Resolving.

File: packages/docs/src/core/docs-map-generator.ts [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+
+  // Default based on common naming
+  if (
+    filename.includes("guide") ||
+    filename.includes("tutorial") ||
+    filename.includes("getting-started")
+  )
+    return "guide";
+  if (filename.includes("api") || filename.includes("reference"))
+    return "reference";
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Addressed in commit 068e1841.  now uses token-boundary matching for  (plus explicit ) instead of raw substring matching, so names like  are no longer false positives. Added regression coverage in  ().

File: packages/docs/src/core/docs-map-generator.ts [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+export async function writeDocsMap(
+  docsMap: DocsMap,
+  options?: { workspaceRoot?: string; docsMapDir?: string }
+): Promise<Result<string, DocsCoreError>> {
+  const workspaceRoot = resolve(options?.workspaceRoot ?? process.cwd());
+  const docsMapDir = resolve(
+    workspaceRoot,
+    options?.docsMapDir ?? DOCS_MAP_DIR
+  );
+  const docsMapPath = join(docsMapDir, DOCS_MAP_FILENAME);
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Follow-up with explicit details: writeDocsMap now validates docsMapDir stays inside workspaceRoot with isPathInsideWorkspace and returns DocsCoreError.validation when it does not. Regression test added in packages/docs/src/__tests__/docs-map-generator.test.ts: returns an error when docsMapDir resolves outside workspace root.

File: packages/docs/src/core/docs-map-generator.ts [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+
+  if (!existsSync(docsMapPath)) {
+    return Result.err(
+      DocsCoreError.validation("docs-map.json not found", {
+        path: docsMapPath,
+      })
+    );
+  }
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-generator.ts]: Implemented. readDocsMap now applies the same workspace boundary validation as writeDocsMap for docsMapDir and returns a validation error when it resolves outside workspaceRoot. Added regression coverage in docs-map-generator.test.ts for the read path.
